File: Prosary/Models/MysteryGroup.swift
//
//  MysteryGroup.swift
//  Prosary
//

import Foundation
import SwiftUI

/// One of the four traditional sets of Rosary mysteries.
enum MysteryGroup: String, Codable, CaseIterable, Identifiable {
  case joyful
  case sorrowful
  case glorious
  case luminous

  var id: String { rawValue }

  /// Display name in English, used as a fallback when no localized name is supplied by the
  /// content layer (e.g. in preset summaries before the real backend is wired up).
  var displayName: String {
    switch self {
    case .joyful:    return String(localized: "mysteryGroup.joyful", defaultValue: "Joyful")
    case .sorrowful: return String(localized: "mysteryGroup.sorrowful", defaultValue: "Sorrowful")
    case .glorious:  return String(localized: "mysteryGroup.glorious", defaultValue: "Glorious")
    case .luminous:  return String(localized: "mysteryGroup.luminous", defaultValue: "Luminous")
    }
  }

  var color: Color {
    switch self {
    case .joyful:    return .adaptive(light: "#1565C0", dark: "#1976D2") // blue
    case .sorrowful: return .adaptive(light: "#6A1B9A", dark: "#7B1FA2") // purple
    case .glorious:  return .adaptive(light: "#C62828", dark: "#D32F2F") // red
    case .luminous:  return .adaptive(light: "#2E7D32", dark: "#388E3C") // green
    }
  }
}

File: Prosary/Models/MysteryGroup.cs
using Windows.UI;

namespace Prosary.Models;

public enum MysteryGroup
{
    Joyful,
    Sorrowful,
    Glorious,
    Luminous
}

public static class MysteryGroupExtensions
{
    /// <summary>English display name, used as a fallback when no localized name is supplied by
    /// the content layer (e.g. in favorite summaries).</summary>
    public static string DisplayName(this MysteryGroup group) => group switch
    {
        MysteryGroup.Joyful => "Joyful",
        MysteryGroup.Sorrowful => "Sorrowful",
        MysteryGroup.Glorious => "Glorious",
        MysteryGroup.Luminous => "Luminous",
        _ => throw new ArgumentOutOfRangeException(nameof(group))
    };

    /// <summary>Accent color for the Home screen's Rosary card — distinct per mystery group, not
    /// the liturgical season color used inside the flow screen's top banner. Matches Android's
    /// <c>MysteryGroup.color</c>.</summary>
    public static Color AccentColor(this MysteryGroup group) => group switch
    {
        MysteryGroup.Joyful => Color.FromArgb(0xFF, 0x15, 0x65, 0xC0), // blue
        MysteryGroup.Sorrowful => Color.FromArgb(0xFF, 0x6A, 0x1B, 0x9A), // purple
        MysteryGroup.Glorious => Color.FromArgb(0xFF, 0xC6, 0x28, 0x28), // red
        MysteryGroup.Luminous => Color.FromArgb(0xFF, 0x2E, 0x7D, 0x32), // green
        _ => throw new ArgumentOutOfRangeException(nameof(group))
    };
}
